components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3311,22 +3325,38 @@ private GroupsGetResponse filterGroupsBySingleAttribute(ExpressionNode node, int
                 groupsList.removeIf(SCIMCommonUtils::isHybridRole);
             }
 
-            if (groupsList != null) {
-                for (String groupName : groupsList) {
-                    if (groupName != null && carbonUM.isExistingRole(groupName, false)) {
-                        // Skip internal roles.
-                        if (CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(groupName) || UserCoreUtil
-                                .isEveryoneRole(groupName, carbonUM.getRealmConfiguration())) {
-                            continue;
-                        }
-                        Group group = getRoleWithDefaultAttributes(groupName, requiredAttributes);
-                        if (group != null && group.getId() != null) {
-                            filteredGroups.add(group);
-                        }
-                    } else {
-                        // Returning null will send a resource not found error to client by Charon.
-                        return new GroupsGetResponse(0, null);
+            int totalGroupCount = groupsList.size();
```

Comment:
Need to set this `totalGroupCount` as the count in the `GroupsGetResponse` 
Check line 3370

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/620/commits/8bab1d14080006b36719860e52034bba4f57397f